src/Microsoft.ComponentDetection.Detectors/pip/PipCommandService.cs

```diff
@@ -131,9 +132,22 @@ public async Task<(PipInstallationReport Report, FileInfo ReportFile)> GenerateI
             pipExecutable,
             null,
             workingDir,
+            cancellationToken,
             pipReportCommand);
 
-        if (command.ExitCode != 0)
+        if (command.ExitCode == -1 && cancellationToken.IsCancellationRequested)
+        {
+            var errorMessage = $"PipReport: Cancelled for file '{formattedPath}' with command '{pipReportCommand.RemoveSensitiveInformation()}'.";
+            using var failureRecord = new PipReportFailureTelemetryRecord
+            {
+                ExitCode = command.ExitCode,
+                StdErr = $"{errorMessage} {command.StdErr}",
+            };
+
+            this.logger.LogDebug("{Error}", errorMessage);
```

Comment:
If we're hitting timeouts we should probably expose this to the user without needing debug logs - maybe just an info log?